utils/realtime.h

```diff
@@ -30,13 +34,18 @@ namespace vulp::utils {
  * \throw std::runtime_error If the operation failed.
  */
 inline void configure_cpu(int cpu) {
+#ifndef __APPLE__
   constexpr int kPid = 0;  // this thread
   cpu_set_t cpuset = {};
   CPU_ZERO(&cpuset);
   CPU_SET(cpu, &cpuset);
   if (::sched_setaffinity(kPid, sizeof(cpu_set_t), &cpuset) < 0) {
     throw std::runtime_error("Error setting CPU affinity");
   }
+#else
+  spdlog::warn(
+      "This function has not been implemented for macOS, doing nothing...");
```

Comment:
```suggestion
      "[utils::configure_cpu] This function does nothing on macOS");
```

Sorry for the back and forth, I should have done a proper review :sweat_smile: